src/runtime_src/core/pcie/common/dmatest.h

```diff
@@ -80,74 +89,73 @@ namespace xcldev {
             }
 
             int result = 0;
-            for_each(threads.begin(), threads.end(), [&](std::future<int> &v) {result += v.get();});
+            std::for_each(threads.begin(), threads.end(), [&](std::future<int> &v) {result += v.get();});
+            return result;
+        }
+
+        void clear() {
+            //Clear out the host shadow buffer
+            std::for_each(mBOList.begin(), mBOList.end(), [&](std::pair<xclBufferHandle, xrt_core::aligned_ptr_type> &v) {
+                    std::memset(v.second.get(), 'x', mSize);
+                });
+        }
+
+        int validate() const {
+            std::unique_ptr<char[]> bufCmp(new char[mSize]);
+            std::memset(bufCmp.get(), pattern, mSize);
```

Comment:
std::vector<char> is probably preferable for this type of RAII:
`std::vector<char> bufCmp(pattern, pattern+mSize);`

Reply:
Updated